components/core/src/LibarchiveFileReader.cpp

```diff
@@ -167,6 +167,60 @@ ErrorCode LibarchiveFileReader::try_read_to_delimiter (char delim, bool keep_del
     return ErrorCode_Success;
 }
 
+ErrorCode LibarchiveFileReader::peek_data_block (size_t size_to_peek, const char*& data_ptr,
```

Comment:
`size_to_peek` seems unnecessary since for sparse files, we could just return a statically sized buffer with nulls. Allowing the caller to dictate how much we peek seems more like a read method which creates unnecessary complexity imo.